File: app/src/main/java/com/kludgenics/cgmlogger/model/treatment/Treatment.java
package com.kludgenics.cgmlogger.model.treatment;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.joda.time.DateTime;

import java.util.Date;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Created by matthiasgranberry on 6/4/15.
 */
public class Treatment extends RealmObject {
    @PrimaryKey
    private String id;
    private Date eventTime;
    private String eventType;
    private String enteredBy;

    private double glucose;
    private String glucoseType;

    private double insulin;
    private String units;

    private String notes;

    private int carbs;
    private int preBolus;

    public Treatment() {
        super();
    }

    public Treatment(String id, Date eventTime, String eventType, String enteredBy, double glucose, String glucoseType, double insulin, String units, String notes, int carbs, int preBolus) {
        this.id = id;
        this.eventTime = eventTime;
        this.eventType = eventType;
        this.enteredBy = enteredBy;
        this.glucose = glucose;
        this.glucoseType = glucoseType;
        this.insulin = insulin;
        this.units = units;
        this.notes = notes;
        this.carbs = carbs;
        this.preBolus = preBolus;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Date getEventTime() {
        return eventTime;
    }

    public void setEventTime(Date eventTime) {
        this.eventTime = eventTime;
    }

    public String getEventType() {
        return eventType;
    }

    public void setEventType(String eventType) {
        this.eventType = eventType;
    }

    public String getEnteredBy() {
        return enteredBy;
    }

    public void setEnteredBy(String enteredBy) {
        this.enteredBy = enteredBy;
    }

    public double getGlucose() {
        return glucose;
    }

    public void setGlucose(double glucose) {
        this.glucose = glucose;
    }

    public String getGlucoseType() {
        return glucoseType;
    }

    public void setGlucoseType(String glucoseType) {
        this.glucoseType = glucoseType;
    }

    public double getInsulin() {
        return insulin;
    }

    public void setInsulin(double insulin) {
        this.insulin = insulin;
    }

    public String getUnits() {
        return units;
    }

    public void setUnits(String units) {
        this.units = units;
    }

    public String getNotes() {
        return notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public int getCarbs() {
        return carbs;
    }

    public void setCarbs(int carbs) {
        this.carbs = carbs;
    }

    public int getPreBolus() {
        return preBolus;
    }

    public void setPreBolus(int preBolus) {
        this.preBolus = preBolus;
    }
}

File: app/src/main/java/com/kludgenics/cgmlogger/model/treatment/Treatment.kt
package com.kludgenics.cgmlogger.model.treatment

import com.kludgenics.alrightypump.cloud.nightscout.records.Treatment
import com.kludgenics.cgmlogger.extension.dateTime
import java.util.Date

import io.realm.RealmObject
import io.realm.annotations.Ignore
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required
import org.joda.time.Duration
import org.joda.time.format.ISODateTimeFormat
import kotlin.properties.Delegates

/**
 * Created by matthiasgranberry on 6/4/15.
 */
open class Treatment : Treatment, RealmObject {

    @PrimaryKey
    override var id: String = ""
    @Required
    override var eventTime: Date = Date()
    override var eventType: String? = null
    override var enteredBy: String? = null

    override var glucose: Double? = null
    override var glucoseType: String? = null

    override var insulin: Double? = null
    override var units: String? = null

    override var notes: String? = null

    override var carbs: Int? = null
    override var preBolus: Int? = null
    override var percent: Int? = null
    override var absolute: Double? = null

    @Ignore
    override var duration: Duration? = null
    open var durationLong: Long? = null
    override var profile: String? = null

    constructor() : super() {
    }

    constructor(treatmentMap: Map<String, String>) {
        duration = if (durationLong != null) Duration(durationLong!!) else null
        Treatment.fromMap(this, treatmentMap)
    }

    constructor(id: String, eventTime: Date, eventType: String, enteredBy: String? = null,
                glucose: Double? = null, glucoseType: String? = null, insulin: Double? = null,
                units: String? = null, notes: String? = null, carbs: Int? = null,
                preBolus: Int? = null, percent: Int? = null, absolute: Double? = null,
                duration: Duration? = null, profile: String? = null) {
        this.id = id
        this.eventTime = eventTime
        this.eventType = eventType
        this.enteredBy = enteredBy
        this.glucose = glucose
        this.glucoseType = glucoseType
        this.insulin = insulin
        this.units = units
        this.notes = notes
        this.carbs = carbs
        this.preBolus = preBolus
        this.percent = percent
        this.absolute = absolute
        this.duration = duration
        this.durationLong = duration?.millis
        this.profile = profile
    }
}

